pkg/hintrunner/zero/zerohint_math.go

```diff
@@ -711,4 +711,72 @@ func createUnsignedDivRemHinter(resolver hintReferenceResolver) (hinter.Hinter,
 	}
 
 	return newUnsignedDivRemHinter(value, div, q, r), nil
+
+}
+
+func newIsQuadResidueHint(x, y hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "IsQuadResidue",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> from starkware.crypto.signature.signature import FIELD_PRIME
+			//>	from starkware.python.math_utils import div_mod, is_quad_residue, sqrt
+			//>	x = ids.x
+			//>	if is_quad_residue(x, FIELD_PRIME):
+			//>		ids.y = sqrt(x, FIELD_PRIME)
+			//>	else:
+			//>		ids.y = sqrt(div_mod(x, 3, FIELD_PRIME), FIELD_PRIME)
+
+			x, err := hinter.ResolveAsFelt(vm, x)
+			if err != nil {
+				return err
+			}
+
+			yAddr, err := y.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			if x.Legendre() == 1 { // If x.Legendre() returns 1 i.e x is a quadratic residue (z ≡ x² (mod q))
+				// calculates the square root
+				xU256 := uint256.Int(x.Bits())
+				xU256.Sqrt(&xU256)
+
+				value := fp.Element{}
+				value.SetBytes(xU256.Bytes())
```

Comment:
Is it possible to use `Sqrt()` function from the `"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"` library? It works directly with the felt values, so you wouldn't need to do the conversion